docs/nx-cloud/concepts/cache-security.md

```diff
@@ -0,0 +1,59 @@
+
+## Potential Malicious Actors
+
+The following section describes different personas that might be malicious actors and how the precautions taken above would counter their nefarious intent.
+
+### A Contractor with Read-Only Access to the Codebase
+
+Because you specified a [read-only token](/nx-cloud/recipes/security/access-tokens) in the `nx.json` file, the contractor would not be able to upload anything to the remote cache. They could use the cache to run tasks quickly, but their actions couldn't affect anyone else's cache.
+
+### A Former Employee that Had Access to a Read-Write Access Token
+
+It is possible to revoke a token and issue a new one, but in practice this is unnecessary. In order to poison the cache, the former employee would need to have both the read-write token and the current code on the latest commit on the `main` branch. The odds of the employee being able to guess the hash value that will be created for the current commit on the `main` branch are infinitesimally small even after a single commit.
+
+### A Developer that Is Able to Create Pull Requests
+
+In an open source environment, anyone in the world can create pull requests so you can't trust the pull request to be safe. This is why we recommend using a [read-write token](/nx-cloud/recipes/security/access-tokens) only for the `main` branch. If you are in a closed source environment where you trust all the developers that have access to create a PR, you could extend that [read-write token](/nx-cloud/recipes/security/access-tokens) to also include PR branches.
+
+### An IT Staffer with Access to a Manually Shared Cache Folder
+
+Nx implicitly trusts the local cache which is stored by default in the `.nx/cache` folder. You can change the location of that folder in the `nx.json` file, so it could be tempting to place it on a network drive and easily share your cache with everyone on the company network. However, by doing this you've voided the guarantee of immutability from your cache. If someone has direct access to the cached files, they could directly poison the cache. Nx will automatically detect if a cache entry has been created in your local cache using a different machine and warn you with an [Unknown Local Cache Error](/recipes/troubleshooting/unknown-local-cache).
+
+### A Hacker that Gained Access to the Nx Cloud Servers
+
+Nx Cloud guarantees your cache entries will remain immutable - once they've been registered they can't be changed. This is guaranteed because the only way to access the cache is through the Nx Cloud API. But what if a hacker were somehow able make their way into the server holding the cache artifacts? Since you set up [end to end encryption](/nx-cloud/recipes/security/encryption), the files they see on disk will be fully encrypted with a key that only exists in your workspace. The worst they could do is mangle the file, making your cached results come back as nonsense.
```

Comment:
we can also say that for the public version we also enforce at the Google Cloud bucket level policies that specifically disable overwrites and deletions of cached artefacts (and this can be enforced by org admins for the on-prem deployments as well).